android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp

```diff
@@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
 {
   return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
 }
+
+JNIEXPORT jstring JNICALL
+Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
+     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
+{
+  Bookmark const * bmk = getBookmark(catId, bmkId);
+  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
```

Comment:
А что значит “из самого РР”? place_page::Info для букмарок по-прежнему недорефакторен и неработоспособен.

> On Mar 25, 2016, at 4:49 PM, Alex Zolotarev <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp <https://github.com/mapsme/omim/pull/2526#discussion_r57444966>:
> 
> > @@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
> >  {
> >    return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
> >  }
> > +
> > +JNIEXPORT jstring JNICALL
> > +Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
> > +     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
> > +{
> > +  Bookmark const * bmk = getBookmark(catId, bmkId);
> > +  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
> Дима, адрес и не должен задаваться ни в каких конструкторах. Вычисление адреса - трудоемкая операция, ее нужно точечно делать только тогда, когда нужно. Но мне не понятно, откуда тогда вызывается этот код. Если из pp, то адрес там должен браться из самого pp, уже вычисленный. А если это для шаринга, то при чем тут букмарка? Посмотри на реализацию. Для вычисления адреса нужна либо точка, либо фича. Это никакого отношения не имеет к букмаркам и не понятно почему вообще тут находится.
> … <x-msg://19/#>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57444966>